lib/backend/syncersv1/updateprocessors/configurationprocessor_test.go

```diff
@@ -0,0 +1,625 @@
+// Copyright (c) 2017 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package updateprocessors_test
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/projectcalico/libcalico-go/lib/apiv2"
+	"github.com/projectcalico/libcalico-go/lib/backend/model"
+	"github.com/projectcalico/libcalico-go/lib/backend/syncersv1/updateprocessors"
+	"github.com/projectcalico/libcalico-go/lib/net"
+	"github.com/projectcalico/libcalico-go/lib/numorstring"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	// What the expected sync datatypes are.
+	isGlobalFelixConfig = iota
+	isNodeFelixConfig
+	isGlobalBgpConfig
+	isNodeBgpConfig
+
+	hostIPMarker = "*HOSTIP*"
+)
+
+var _ = Describe("Test the generic configuration update processor and the concrete implementations", func() {
+	// Define some common values
+	perNodeFelixKey := model.ResourceKey{
+		Kind: apiv2.KindFelixConfiguration,
+		Name: "node.mynode",
+	}
+	globalFelixKey := model.ResourceKey{
+		Kind: apiv2.KindFelixConfiguration,
+		Name: "default",
+	}
+	invalidFelixKey := model.ResourceKey{
+		Kind: apiv2.KindFelixConfiguration,
+		Name: "foobar",
+	}
+	globalClusterKey := model.ResourceKey{
+		Kind: apiv2.KindClusterInformation,
+		Name: "default",
+	}
+	nodeClusterKey := model.ResourceKey{
+		Kind: apiv2.KindClusterInformation,
+		Name: "node.mynode",
```

Comment:
i'm not intending it to be valid - but that validation will happen in the client code - so i'm really using it here to get the test coverage up since it's a simple resource type and has a field with a special stringifier.